src/abstractdataframe/io.jl

```diff
@@ -92,16 +99,72 @@ function html_escape(cell::AbstractString)
     return cell
 end
 
-Base.show(io::IO, mime::MIME"text/html", df::AbstractDataFrame; summary::Bool=true) =
-    _show(io, mime, df, summary=summary)
+"""
+    show(io::IO, mime::MIME, df::AbstractDataFrame;
+         allrows::Bool = !get(io, :limit, false),
+         allcols::Bool = !get(io, :limit, false),
+         summary::Bool=true)
+
+Render a data frame to an I/O stream in MIME type `mime`.
+
+# Arguments
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `mime::MIME`: supported MIME types are: `"text/plain"`, `"text/html"`, `"text/latex"`,
+  `"text/csv"`, `"text/tab-separated-values"`
+- `df::AbstractDataFrame`: The data frame to print.
+- `allrows::Bool `: supported only for plain text, HTML and LaTeX MIME types;
+   whether to print all rows, rather than a default number defined in
+   `HTML_NROWS` and `LATEX_NROWS` constants for these MIME types.
+- `allcols::Bool`: supported only for HTML and LaTeX mime types;
+   whether to print all columns, rather than a default number defined in
+   `HTML_NCOLS` and `LATEX_NCOLS` constants for these MIME types.
+- `summary::Bool = true`: supported only for HTML mime type;
+   Whether to print a brief string summary of the data frame.
+
+# Examples
+```jldoctest
+julia> show(stdout, MIME("text/latex"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+\\begin{tabular}{r|cc}
+        & A & B\\\\
+        \\hline
+        & Int64 & String\\\\
+        \\hline
+        1 & 1 & x \\\\
+        2 & 2 & y \\\\
+        3 & 3 & z \\\\
+\\end{tabular}
+14
+
+julia> show(stdout, MIME("text/csv"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+"A","B"
+1,"x"
+2,"y"
+3,"z"
+```
+"""
+Base.show(io::IO, mime::MIME"text/html", df::AbstractDataFrame;
```

Comment:
Can you use a broader signature so that only one method is needed for different MIME types? That will also allow attaching the docstring to the right method. Then also move it to the top of the file.

Reply:
Done. Now the default fallback will be to `"text/plain"` if some unknown mime type is passed.

Reply:
I have removed one unused method for `printtable` (probably it was exported in the past?)

Reply:
> Now the default fallback will be to `"text/plain"` if some unknown mime type is passed.

Shouldn't this be an error instead?

Reply:
OK - I will make it an error.